source/Calamari/Util/Environments/EnvironmentHelper.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using Calamari.Integration.FileSystem;
+using System.Runtime.InteropServices; // Needed for dotnetcore
+#if VISUALBASIC_SUPPORT
+using Microsoft.VisualBasic.Devices;
+#endif
+
+namespace Calamari.Util.Environments /* This is in the 'Environments' namespace to avoid collisions with CrossPlatformExtensions - long story */
+{
+    public static class EnvironmentHelper
+    {
+        public static string[] SafelyGetEnvironmentInformation()
+        {
+            var envVars = new List<string>();
+            SafelyAddEnvironmentVarsToList(ref envVars);
+            SafelyAddPathVarsToList(ref envVars);
+            SafelyAddProcessVarsToList(ref envVars);
+            SafelyAddComputerInfoVarsToList(ref envVars);
+            return envVars.ToArray();
+        }
+
+        static void SafelyAddEnvironmentVarsToList(ref List<string> envVars)
+        {
+            try
+            {
+#if NET40
+                envVars.Add($"OperatingSystem: {Environment.OSVersion.ToString()}");
+                envVars.Add($"OsBitVersion: {(Environment.Is64BitOperatingSystem ? "x64" : "x86")}");
+                envVars.Add($"Is64BitProcess: {Environment.Is64BitProcess.ToString()}");
+#else
+                envVars.Add($"OperatingSystem: {System.Runtime.InteropServices.RuntimeInformation.OSDescription.ToString()}");
+                envVars.Add($"OsBitVersion: {System.Runtime.InteropServices.RuntimeInformation.OSArchitecture.ToString()}");
+#endif
+                envVars.Add($"CurrentUser: {System.Security.Principal.WindowsIdentity.GetCurrent().Name}");
+                envVars.Add($"MachineName: {Environment.MachineName}");
+                envVars.Add($"ProcessorCount: {Environment.ProcessorCount.ToString()}");
+            }
+            catch
+            {
+                // silently fail.
+            }
+        }
+
+        static void SafelyAddPathVarsToList(ref List<string> envVars)
+        {
+            try
+            {
+                envVars.Add($"CurrentDirectory: {Directory.GetCurrentDirectory()}");
+                envVars.Add($"TempDirectory: {Path.GetTempPath()}");
+            }
+            catch
+            {
+                // silently fail.
+            }
+        }
+
+        static void SafelyAddProcessVarsToList(ref List<string> envVars)
+        {
+            try
+            {
+                envVars.Add($"HostProcessName: {Process.GetCurrentProcess().ProcessName}");
+            }
+            catch
+            {
+                // silently fail.
+            }
+        }
+
+        static void SafelyAddComputerInfoVarsToList(ref List<string> envVars)
```

Comment:
We really shouldn't use `ref`, pretty much ever. I think remove these... but a better long-term change would be to make this method, and its brothers/sisters into:

```
static IEnumerable<string> SafelyGetComputerInfo()
{
  try
  {
    var computerInfo = new ComputerInfo();
    yield return $"TotalPhysicalMemory: {computerInfo.TotalPhysicalMemory.ToFileSizeString()}"
    yield return $"AvailablePhysicalMemory: {computerInfo.AvailablePhysicalMemory.ToFileSizeString()}"
  }
  catch
  {
    // silently fail
  }
}
```